pyqt-apps/siriushla/as_ap_sofb/graphics/correctors.py

```diff
@@ -0,0 +1,123 @@
+        self.updater[0].some_changed('ref', 'Zero')
+        self.updater[1].some_changed('val', 'Kicks')
+        self.updater[1].some_changed('ref', 'Zero')
+
+        self.add_kicklimits_curves()
+
+    def add_kicklimits_curves(self):
+        grpbx = QGroupBox('Show Kick Limits', self)
+        vbl = QVBoxLayout(grpbx)
+        self.hbl.addWidget(grpbx)
+        self.hbl.addStretch(1)
+        chcbox1 = QCheckBox('Kicks', grpbx)
+        chcbox2 = QCheckBox('Delta Kicks', grpbx)
+        chcbox1.setChecked(True)
+        chcbox2.setChecked(True)
+        vbl.addWidget(chcbox1)
+        vbl.addWidget(chcbox2)
+
+        chcboxs = (chcbox1, chcbox2)
+        names = ('Max Kicks', 'Max dKickx')
+        pvs = ('MaxKick', 'MaxDeltaKick')
+        stys = (4, 2)
+        wids = (3, 2)
+        plns = ('x', 'y')
+        corrs = ('CH', 'CV')
+        for chb, name, pvi, sty, wid in zip(chcboxs, names, pvs, stys, wids):
+            pen = mkPen(QColor(0, 0, 0))
+            pen.setStyle(sty)
+            pen.setWidth(wid)
+            for pln, corr in zip(plns, corrs):
+                maxkick = InfLine(
+                    conv=1e-6, pos=0.0, pen=pen, angle=0, name=name)
+                minkick = InfLine(conv=-1e-6, pos=0.0, pen=pen, angle=0)
+                self.graph[pln].addItem(maxkick)
+                self.graph[pln].addItem(minkick)
+                chan = SiriusConnectionSignal(self.prefix + pvi + corr + '-RB')
+                self._chans.append(chan)
+                chan.new_value_signal[float].connect(maxkick.setValue)
+                chan.new_value_signal[float].connect(minkick.setValue)
+                chb.toggled.connect(maxkick.setVisible)
+                chb.toggled.connect(minkick.setVisible)
+
+    def channels(self):
+        chans = super().channels()
+        chans.extend(self._chans)
+        return chans
+
+    @staticmethod
+    def get_default_ctrls(prefix):
+        chans = [
+            SiriusConnectionSignal(prefix+'DeltaKicksCH-Mon'),
+            SiriusConnectionSignal(prefix+'DeltaKicksCV-Mon'),
+            SiriusConnectionSignal(prefix+'KicksCH-Mon'),
+            SiriusConnectionSignal(prefix+'KicksCV-Mon')]
+        ctrls = {
+            'Delta Kicks': {
+                'x': {
+                    'signal': chans[0].new_value_signal,
+                    'getvalue': chans[0].getvalue},
+                'y': {
+                    'signal': chans[1].new_value_signal,
+                    'getvalue': chans[1].getvalue}},
+            'Kicks': {
+                'x': {
+                    'signal': chans[2].new_value_signal,
+                    'getvalue': chans[2].getvalue},
+                'y': {
+                    'signal': chans[3].new_value_signal,
+                    'getvalue': chans[3].getvalue}},
+            }
+        return chans, ctrls
+
+
+class InfLine(InfiniteLine):
+
+    def __init__(self, conv=1, pos=None, **kwargs):
+        if pos is not None:
+            pos *= conv
```

Comment:
Is this `pos` really used?

Reply:
Thanks @anacso17! Yes it is needed, I just forgot to pass it to the `super().__init__` call.